scipy/stats/_survival.py

```diff
@@ -374,3 +384,122 @@ def _ecdf_right_censored(sample):
     sf = np.cumprod((n - d) / n)
     cdf = 1 - sf
     return t, cdf, sf, n, d
+
+
+@dataclass
+class LogRankResult:
+    statistic: np.ndarray
+    pvalue: np.ndarray
+
+
+def _at_risk(times: np.ndarray, all_times: np.ndarray) -> np.ndarray:
+    """Times need to be sorted."""
+    at_risk = []
+    n_times = len(times)
+    for x in all_times:
+        idx_ = bisect.bisect_left(times, x)  # type: ignore[call-overload]
+        at_risk.append(n_times-idx_)
+    return np.array(at_risk)
+
+
+def _iv_log_rank(
+    sample: CensoredData, control: CensoredData,
+) -> tuple[CensoredData, CensoredData]:
+    if not (isinstance(sample, CensoredData) and
+            isinstance(control, CensoredData)):
+        raise ValueError(
+            ""
+        )
+
+    return sample, control
+
+
+def log_rank(sample: CensoredData, control: CensoredData) -> LogRankResult:
+    """Log Rank test.
+
+    Parameters
+    ----------
+    sample, control : CensoredData
+        Sample and control data to compare based on their survival functions.
+        The order does not matter, `sample` and `control` can be interverted.
```

Comment:
If the order doesn't matter, then let's avoid the names `sample` and `control`. Then we don't even need to state that the order doesn't matter because there is no reason to suspect that the order matters. The most common names for two arguments are `x, y`.